tests/unit/s2n_connection_test.c

```diff
@@ -283,6 +290,134 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_connection_free(conn));
     }
 
+    /* Test: signature algorithm and hash can be retrieved after the handshake.
+     * Check both TLS1.2 and TLS1.3, because they use different signature negotiation logic.
+     * Check for both the server and client certificates, because they use different negotiation logic.
+     */
+    {
+        int (*sig_alg_getters[])(struct s2n_connection *, s2n_tls_signature_algorithm *) = {
+                s2n_connection_get_selected_signature_algorithm,
```

Comment:
Does this array make the code that much cleaner? TBH I'm not sure the cleanliness is worth the extra effort to it takes to understand this test.